src/watchman/api_v1_handler.clj

```diff
@@ -65,7 +66,13 @@
       (do
         (models/remove-host-from-role (:id host) (:id role))
         {:status 204}) ; No content
-      (create-json-error-response 404 "hostname not found in role"))))
+      (create-json-error-response 404 "hostname not found in role")))
+
+  (POST "/snooze" {:keys [params role]}
+    (let [snooze-duration (-> params :duration Long/parseLong)
+          role-id (:id role)]
+      (models/snooze-role role-id snooze-duration)
+      (redirect (str "/roles/" role-id)))))
```

Comment:
I think these API routes shouldn't be redirecting to HTML pages. This should just acknowledge the snooze has been set with a 200, or maybe a `204: No Content`.